pkg/ipfs/client.go

```
@@ -138,6 +141,23 @@ func (cl Client) SwarmAddresses(ctx context.Context) ([]string, error) {
 	return addresses, nil
 }
 
+// SwarmConnect establishes concurrent connections to each peer from the provided `peers` list.
+// It spawns a goroutine for each peer connection. In the event of a connection failure,
+// a warning log containing the error and peer details is generated.
+func (cl Client) SwarmConnect(ctx context.Context, peers []peer.AddrInfo) {
+	var wg sync.WaitGroup
+	for _, p := range peers {
+		wg.Add(1)
+		go func(ctx context.Context, p peer.AddrInfo) {
+			defer wg.Done()
+			if err := cl.API.Swarm().Connect(ctx, p); err != nil {
+				log.Ctx(ctx).Warn().Err(err).Stringer("peer", p).Msg("failed to connect to peer")
```

We might want to return an error here and let the caller handle it, thoughts?